src/ops/QueryOps.ts

```diff
@@ -116,4 +119,89 @@ export class QueryOps extends BaseOp {
+                reject(new ThingIFError(Errors.ArgumentError, "QueryGroupedHistoryStatesRequest instance is null"));
+            } else if (!groupedQuery.alias) {
+                reject(new ThingIFError(Errors.ArgumentError, "alias is null or empty"));
+            } else if (!KiiUtil.isString(groupedQuery.alias)) {
+                reject(new ThingIFError(Errors.ArgumentError, "alias is not string"));
+            } else if (!groupedQuery.range) {
+                reject(new ThingIFError(Errors.ArgumentError, "range is null"));
+            } else if (!KiiUtil.isObject(groupedQuery.range)) {
+                reject(new ThingIFError(Errors.ArgumentError, "range is not object"));
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    private validateAggregation(aggregation: Aggregation): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            if (!aggregation) {
+                reject(new ThingIFError(Errors.ArgumentError, "Aggregation instance is null"));
+            } else if (!aggregation.field) {
+                reject(new ThingIFError(Errors.ArgumentError, "field of aggregation is null or empty"));
+            } else if (!KiiUtil.isString(aggregation.field)) {
+                reject(new ThingIFError(Errors.ArgumentError, "field of aggregation is not string"));
+            } else if (!aggregation.fieldType) {
+                reject(new ThingIFError(Errors.ArgumentError, "fieldType of aggregation is null or empty"));
+            } else if (!KiiUtil.isString(aggregation.fieldType)) {
+                reject(new ThingIFError(Errors.ArgumentError, "fieldType of aggregation is not string"));
+            } else if (!aggregation.type) {
+                reject(new ThingIFError(Errors.ArgumentError, "type of aggregation is null or empty"));
+            } else if (!KiiUtil.isString(aggregation.type)) {
+                reject(new ThingIFError(Errors.ArgumentError, "type of aggregation is not string"));
+            } else {
+                if (aggregation.type === FunctionType.MAX ||
+                    aggregation.type === FunctionType.MEAN ||
+                    aggregation.type === FunctionType.MIN ||
+                    aggregation.type === FunctionType.SUM) {
+                    if (aggregation.fieldType !== FieldType.DECIMAL &&
+                        aggregation.fieldType !== FieldType.INTEGER) {
+                        reject(new ThingIFError(Errors.ArgumentError, `unsupported field type ${aggregation.fieldType} for ${aggregation.type} aggregation`));
+                        return;
+                    }
+                }
+                resolve();
+            }
+        });
+    }
+
+    aggregateQuery(requestObject: AggregateGroupedHistoryStatesRequest): Promise<Array<AggregatedResults>> {
+        return new Promise<Array<AggregatedResults>>((resolve, reject) => {
+            this.validateGroupedQuery(requestObject.groupedQuery).then(() => {
```

Comment:
@shinsuke-nara for js, developer can pass non `AggregateGroupedHistoryStatesRequest` instance (like string, some other object and etc.) to aggregateQuery API, so in JSSDK, we validate parameters in `**Ops`